src/librustc_data_structures/profiling.rs

```diff
@@ -99,10 +99,12 @@ use parking_lot::RwLock;
 
 /// MmapSerializatioSink is faster on macOS and Linux
 /// but FileSerializationSink is faster on Windows
-#[cfg(not(windows))]
+#[cfg(all(not(windows), not(target_arch = "wasm32")))]
 type SerializationSink = measureme::MmapSerializationSink;
-#[cfg(windows)]
+#[cfg(all(windows, not(target_arch = "wasm32")))]
 type SerializationSink = measureme::FileSerializationSink;
+#[cfg(target_arch = "wasm32")]
```

Comment:
When using wasi, the `FileSerializationSink` should be used.

Reply:
(this is `#[cfg(target_os = "wasi")]`)